### DISCUSSION TITLE
Possible to validate extra requirements to optional field, in case a value is supplied?

### DISCUSSION THREAD

**Discussion by ollebergkvist** [open-circle/valibot#503] [Q&A] [answered]
I'm working on a validation requirement where a postal code (postalcode) is considered optional if no value is supplied (an empty string). However, if a value is supplied, it must fulfill certain requirements.

postalcode: optional(
    number([
      length(5),
      custom((value) => /^\d+$/.test(value), 'Postal code must be 5 digits'),
    ])
  ),

This fails for empty strings:

```
next:dev:   issues: [
next:dev:     {
next:dev:       reason: 'string',
next:dev:       validation: 'length',
next:dev:       origin: 'value',
next:dev:       message: 'Invalid length',
next:dev:       input: '',
next:dev:       path: [Array],
next:dev:       abortEarly: undefined,
next:dev:       abortPipeEarly: undefined,
next:dev:       skipPipe: undefined
next:dev:     },
next:dev:     {
next:dev:       reason: 'string',
next:dev:       validation: 'custom',
next:dev:       origin: 'value',
next:dev:       message: 'Postalcode can only contain numerical values',
next:dev:       input: '',
next:dev:       path: [Array],
next:dev:       abortEarly: undefined,
next:dev:       abortPipeEarly: undefined,
next:dev:       skipPipe: undefined
next:dev:     }
next:dev:   ]
next:dev: }
``

**Comment by fabian-hiller** [comment, marked as answer]
Yes, this is possible but I would write it differently. Feel free to try it out in our [playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6AK6tgbABQBtenHHkYfKGVmFCASgA0S8RyjBWAcwUYxpPuZg9ZAVj1EAChA5k4LACZ84IBIccBT+jnA+wGbAMKi6LlZ+TKCaOgC6+vQZDMxswVB8qBKk8KJWqMjYfK7IUKh8stz8gi6EAIwATADMACyOujks7BA21hAWBUUlOrRAA).

```ts
import * as v from 'valibot';

const Schema = v.union([
  v.literal(''),
  v.string([v.length(5, 'Postal code must be 5 digits'), v.decimal()]),
]);
```

**Reply by ollebergkvist** [reply]
Amazing, thank you!
